crates/polars-io/src/file_cache/utils.rs

```diff
@@ -69,13 +69,27 @@ pub fn init_entries_from_uri_list<A: AsRef<[Arc<str>]>>(
         .unwrap_or_else(get_env_file_cache_ttl);
 
     if is_cloud_url(first_uri) {
-        let (_, object_store) = pl_async::get_runtime()
-            .block_on_potential_spawn(build_object_store(first_uri, cloud_options))?;
-        let object_store = PolarsObjectStore::new(object_store);
+        let object_stores = pl_async::get_runtime().block_on_potential_spawn(async {
+            futures::future::try_join_all(
+                (0..if first_uri.starts_with("http") {
+                    // Object stores for http are tied to the path.
+                    uri_list.len()
```

Comment:
Without this, scanning a list of HTTP paths would just scan the first file N timess.